### ISSUE TITLE
Possible bug relating to items being erased from chests?

### ISSUE THREAD

**Issue by GalacticRuler** [abearcodes/Valheim#21]
https://github.com/robclancy/valheim-bigger-chests/issues/1 See here

**Comment by abearcodes** [comment]
That seems like something that totally shouldn't happen if things are working right... You have quite the collection of mods, so it's going to be hard to understand what is actually causing it. That said, I definitely think Biggest Chests has nothing to do with this as it does not alter crafting. 

Most likely, if you disable/uninstall Crafting With Containers, it will fix the issue as nothing will be checking the rest of the chests except your inventory. A guaranteed option to really figure out what is going on is to enable Crafting With Containers and then start adding mods one by one until the issue happens again. 

Also, in the other issue you mention crafting with containers reporting 0 items used. Can you please try enabling debugging, try to replicate the issue and once you know that it happened share your Player.log?

To enable debug logging in BepInEx, open the file at `.\Valheim\BepInEx\config\BepInEx.cfg` and under the section "[Logging.Console]" set "LoggingLevels" (around line 60) to "All" like so:
```
[Logging.Console]
...
## Which log levels to show in the console output.
# Setting type: LogLevel
# Default value: Fatal, Error, Warning, Message, Info
# Acceptable values: None, Fatal, Error, Warning, Message, Info, Debug, All
# Multiple values can be set at the same time by separating them with , (e.g. Debug, Warning)
LogLevels = All
```

Restart your game and you should now get debug output from all mods.

Once you manage to replicate the issue (I don't know how ofc, just try to do what you did before), please post your Player.log under this issue. You can find the Player.log under `C:\Users\<username>\AppData\LocalLow\IronGate\Valheim`

**Comment by GalacticRuler** [comment]
will report back if it happens again

thanks for the info on how to switch on higher level logging and where to go!

**Comment by Ballig** [comment]
Happened to me with Surtling cores. Had 18 in a chest - removed 10 from the chest into my inventory to create one kiln and one smelter (require 5 each). After I made them I had 5 cores in my inventory and no cores left in my chest. So it used 13 Surtling cores to make two items that require 5 each. It may be related to having items in your inventory and also your nearby chests when you are building.

**Comment by VirtueAvatar** [comment]
This has happened to me and a couple of others as well, specifically with surtling cores. The cores seem to be safe in our chests so long as we don't place anything in the same crafting station area that doesn't use the cores and the chests - we've placed and dismantled all of those buildings around the time we lost the cores. 

We're using:

AddAllFuel
AutoRepair
BlastFurnaceFix
Build Camera
ClockMod
ColourfulSigns
DiscardInventoryItem
ExploreTogether
FermenterStatus
FloatingItems
HoldAttack
ImNoMasochist
ItsJustWood
MinimalStatusEffects
modify_data_rate
multiplayerboatdamage
NexusUpdate
NoStamCosts
NotSoNeedyCraftingStation
SkipIntro
QuickConnect
QuickDeposit
RememberIPDamnit
RemoveDeathPins
ShowPlantProgress
SleepWithoutSpawn
TorchesEternal
TripleBronze
Valheim.DisplayBepInExInfo
Valheim BiggerChests

**Comment by Ballig** [comment]
Okay, I can reproduce it.

The Craft From Containers mod takes materials from your CHESTS first, not your inventory. If you try to craft something and don't have enough materials in your chest, it then takes it from your inventory. BUT it will first eat the materials you had left in your chest (that weren't enough to build the item).

Tested this with wood. I had 34 wood total. I took 29 wood out of the chests and left 5 wood in the chest. I then made a chest that requires 10 wood. I should have had 24 wood after making the chest (34 - 10) but instead I only had 19 wood remaining. It checked my chest for wood and found 5, which wasn't enough to make the chest, but then ATE that wood before taking the full 10 from my inventory (instead of taking 5 from my chest and 5 from my inventory).

**Comment by abearcodes** [comment]
In 1.0.8 I re-introduced the ability to take from the players' inventory first. But, I typoed on one of the checks and it ended up taking the items twice.

I'm sorry for the items of those 300 people that downloaded that version. Rushing and copy-pasting bits of old code while re-introducing the feature caused this :(

This was the bug: https://github.com/abearcodes/Valheim/commit/239b37adbb571d653805ba2409698b1a7f9e1cb9

If TakeFromPlayerInventoryFirst was disabled (which it is by default), it would take the needed amount from the surrounding chests, and then additionally from the player's inventory.

As of 1.0.9 (and how was intended for 1.0.8), with the re-introduction of the `TakeFromPlayerInventoryFirst` option enabled, the way the plugin works is:
1. if `TakeFromPlayerInventoryFirst` enabled, it tries to take as much as possible from the player inventory, counts down amount of needed taken
2. iterates containers and for each of them, tries to take items from the inventory, counts down the amount of items needs
3. if `TakeFromPlayerInventoryFirst` **disabled**, it tries to take as much as possible from the player inventory, counts down amount of needed taken

**Comment by abearcodes** [comment]
Please let me know if you are still encountering issues with version 1.0.9, I'll keep this issue open for now and will close it in about a week once there's no new reports on this.

**Comment by GalacticRuler** [comment]
So far nothing has vanished!  thanks for the fast work and wonderful mod ^^

**Comment by VirtueAvatar** [comment]
This was resolved in 1.0.9. Thanks!

**Comment by abearcodes** [comment]
Ah okay, then I’m closing the issue and if it resurfaces just open a new one as the cause will most likely be unrelated to what happened in 1.0.8. Thanks everyone for your input
